src/tls.c

```diff
@@ -91,11 +92,18 @@ enftun_tls_load_credentials(struct enftun_tls* tls,
     }
     enftun_log_debug("Loaded client TLS certificate %s\n", cert_file);
 
-    if (!(SSL_CTX_use_PrivateKey_file(tls->ctx, key_file, SSL_FILETYPE_PEM) ||
-          SSL_CTX_use_PrivateKey_file(tls->ctx, key_file, SSL_FILETYPE_ASN1)))
+    if (enftun_tpm2tss_is_tss_key(key_file))
     {
-        enftun_log_ssl_error("Failed to load client TLS key %s:", key_file);
-        goto err;
+        enftun_tpm2tss_ssl_ctx_set(cert_file, tls->ctx);
```

Comment:
Need to handle an erro rand a similar error message here as in the PEM and ASN1 case:

```c
if (!enftun_tls_tpm_use_key(tls, key_file))
{
    enftun_log_error("Failed to load client TLS key from TPM %s", key_file);
    goto err;
}
```

Reply:
`cert_file` should be `key_file`.

Reply:
This section has been restructured to account for XTT incompatibilities.